### PR TITLE
Add SONiC CLI

### PR DESCRIPTION
**Pull request by jleveque**
- Add SONiC CLI as a Python package
- Debian package is now built using Python setuptools instead of dpkg-buildpackage
- Create sonic_eeprom and sonic_sfp Python packages to cleanly house modules
- Merge changes from internal repo

### CODE REVIEW

File: scripts/decode-syseeprom [hunk 1]
```diff
@@ -15,34 +15,52 @@ try:
...
+# Returns platform model (platform string) and submodel (HW SKU)
+def get_platform():
+    try:
+        proc = subprocess.Popen([SONIC_CFGGEN, '-v', PLATFORM_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        model = stdout.rstrip('\n')
```
Comment (stcheng) [inline scripts/decode-syseeprom]: rename model and submodel to platform and hwsku?

File: scripts/portstat [hunk 2]
```diff
@@ -66,7 +66,6 @@ filter_bucket_dict = {
     'RFCR': 1,
     'RJBR': 1,
     'RDBGC0': 2,
-    'RDBGC2': 2,
```
Comment (stcheng) [inline scripts/portstat]: Why is this line removed?

File: scripts/portstat [hunk 3]
```diff
@@ -310,7 +309,7 @@ Examples:
             sys.exit(0)
 
     try:
-        bcmcmd_output = subprocess.Popen((['/usr/sbin/bcmcmd',
+        bcmcmd_output = subprocess.Popen((['/usr/bin/bcmcmd',
```
Comment (stcheng) [inline scripts/portstat]: define the path at the top of the script?

File: sonic_sfp/sfputilbase.py [hunk 4]
```diff
@@ -5,22 +5,22 @@
...
+    import getopt
+    import re
+    import bcmshell
+    import pprint
+    from math import log10
+    from sonic_eeprom import eeprom_dts
+    from sff8472 import sff8472InterfaceId
+    from sff8472 import sff8472Dom
+    from sff8436 import sff8436InterfaceId
+    from sff8436 import sff8436Dom
```
Comment (stcheng) [inline sonic_sfp/sfputilbase.py]: move sff8436/sff8472 to sonic_sfp folder?

File: scripts/portstat (hunk 2 above)
Comment (jleveque) [inline scripts/portstat]: This was a change I merged from our internal repo. You removed this, claiming it was an unnecessary drop counter on 5/12/2016.

File: scripts/decode-syseeprom (hunk 1 above)
Comment (jleveque) [inline scripts/decode-syseeprom]: Resolved: https://github.com/Azure/sonic-utilities/pull/7/commits/06619a6b74ccc05c90e4c278735e870ffdd2b77b

File: scripts/portstat (hunk 3 above)
Comment (jleveque) [inline scripts/portstat]: Resolved: https://github.com/Azure/sonic-utilities/pull/7/commits/05b0c7d259f77751d5f0cd252f8aa7e971718cd6

File: sonic_sfp/sfputilbase.py (hunk 4 above)
Comment (jleveque) [inline sonic_sfp/sfputilbase.py]: Already done. Since sfputilbase, sff8436 and sff8472 all reside in the same package, they can reference one another using relative paths. I could make the paths absolute for clarity, if desired.

**Comment by lguohan** [comment]
:shipit:

**Comment by stcheng** [comment]
Please follow the GitHub Workflow and update the commit title and message.
Thanks! 😃
